go/vt/vtgate/evalengine/integration/comparison_test.go

```diff
@@ -519,3 +519,19 @@ func TestCharsetConversionOperators(t *testing.T) {
 		}
 	}
 }
+
+func TestNumericFuncFloor(t *testing.T) {
+	var numbers = []string{
+		`0`, `-1`, `1`, `0.0`, `1.0`, `-1.0`, `1.0E0`, `-1.0E0`, `0.0E0`,
+		strconv.FormatFloat(math.MaxFloat64, 'E', -1, 64),
+		strconv.FormatFloat(math.MaxFloat32, 'E', -1, 32),
+		`NULL`, `1.10aa`, `-1.10aa`, `0.1.99`, `-0.1.99`, `.99`,
+	}
+
+	var conn = mysqlconn(t)
+	defer conn.Close()
+
+	for _, number := range numbers {
+		compareRemoteQuery(t, conn, fmt.Sprintf(" SELECT FLOOR('%s')", number))
```

Comment:
This integration test is broken: you're always wrapping the input to `FLOOR` with ticks, so you're always trying to floor string values. You need to remove those ticks (and place them in some of the values in the `numbers` slice) so you're actually flooring numbers. Then you'll see that your implementation is not correct for many different numeric types.